deltachat-ffi/src/lib.rs

```diff
@@ -1407,6 +1409,37 @@ pub unsafe extern "C" fn dc_set_chat_profile_image(
         .unwrap_or(0)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn dc_set_chat_mute_duration(
+    context: *mut dc_context_t,
+    chat_id: u32,
+    duration: i64,
+) -> libc::c_int {
+    if context.is_null() || chat_id <= constants::DC_CHAT_ID_LAST_SPECIAL as u32 {
```

Comment:
You only need to check `context.is_null()` here.  The special chat id bit is taken care off by the rust api, the `.unwrap_or_log_default()` bit a few lines below will take care of handling this error nicely (nicer than this as it logs normally and provides a better message).